### PR TITLE
B-20758 TPPS Paid Payment Request Status Update INT

### PR DESCRIPTION
**Pull request by traskowskycaci**
## Summary

When we process the TPPS Paid Invoice Report, we also want to update the status of all the payment requests in the report itself to have a status of PAID.

Each row in the TPPS Paid Invoice Report refers to one payment service item - this means we can have many rows that all refer to the same payment request.

Given we only need to update the payment request status once, the logic for updating the payment request status keeps track of which payment requests that have already updated with a `map`. For each row of the TPPS report, we find the corresponding payment request - then we check if that payment request is in our map already. If it is, that means we already updated the status, so we don't need to do it again.

Work in this PR:
- added logic for updating payment request status to PAID
- added logging to give specific information that mentions the payment request number, payment request status, and TPPS data invoice number that was updated (_payment request number matches invoice number_)
- added a new test specifically for checking payment request status in pkg/services/invoice/process_tpps_paid_invoice_report_test.go
- also cleaned up the other tests in that file (didn't need to be calling `factory.BuildPaymentRequestToInterchangeControlNumber` for any of the test payment requests, was initially a copy/paste from other EDI processing tests) 

Note:
The second AC mentioning seeing the "TPPS PAID" displayed next to the payment request number on the UI was already covered/met by [B-20674](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20674)

### How to test

1. This is another of those "look really closely at the tests" PRs. Feel free to debug and check it out though!
2. Here are some screenshots to show what we are doing with the map logic
3. Here is what a parsed TPPS report looks like - note the _duplicate_ InvoiceNumbers on the right. The invoice_number for each TPPS report is how we find the matching payment request - it will match the payment request's payment_request_number.

image

4. Here is the map of the payment request with key:value pairs of id:payment_request_number. These show that we only did 4 transactions to update the payment request status once for each payment request in the TPPS report - more efficient than updating the payment request status when we don't need to.

image

### CODE REVIEW

**Review by danieljordan-caci — APPROVED** [review]
LGTM!

File: pkg/services/invoice/process_tpps_paid_invoice_report.go [hunk 1]
```diff
@@ -70,10 +70,54 @@ func (t *tppsPaidInvoiceReportProcessor) ProcessFile(appCtx appcontext.AppContex
...
+				// to keep track of which payment requests we've already updated the status to PAID for and
+				// only update it's status once, using a map to keep track of already updated payment requests
+				_, paymentRequestExistsInUpdatedStatusMap := paymentRequestWithStatusUpdatedToPaid[paymentRequest.ID.String()]
+				if !paymentRequestExistsInUpdatedStatusMap {
+					paymentRequest.Status = models.PaymentRequestStatusPaid
+					err = txnAppCtx.DB().Update(&paymentRequest)
+					if err != nil {
+						txnAppCtx.Logger().Error("failure updating payment request to PAID", zap.Error(err))
+						return fmt.Errorf("failure updating payment request status to PAID: %w", err)
+					}
```
Comment (cameroncaci) [inline pkg/services/invoice/process_tpps_paid_invoice_report.go:105]: Could you update this log to include payment request ID?

File: pkg/services/invoice/process_tpps_paid_invoice_report.go [hunk 2]
```diff
@@ -83,6 +127,14 @@ func (t *tppsPaidInvoiceReportProcessor) EDIType() models.EDIType {
 	return models.TPPSPaidInvoiceReport
 }
 
+func (t *tppsPaidInvoiceReportProcessor) logTPPSInvoiceReportWithPaymentRequest(appCtx appcontext.AppContext, tppsResponse tppsReponse.TPPSData, paymentRequest models.PaymentRequest) {
+	appCtx.Logger().Info("TPPS Paid Invoice Report log",
+		zap.String("PaymentRequestNumber", paymentRequest.PaymentRequestNumber),
+		zap.String("PaymentRequest.Status", string(paymentRequest.Status)),
+		zap.String("TPPSPaidInvoiceReportEntry.InvoiceNumber", tppsResponse.InvoiceNumber),
+	)
```
Comment (cameroncaci) [inline pkg/services/invoice/process_tpps_paid_invoice_report.go:135]: Could we add the UUID too?

**Review by cameroncaci — CHANGES_REQUESTED** [review]

File: pkg/services/invoice/process_tpps_paid_invoice_report.go (hunk 2 above)
Comment (traskowskycaci) [reply pkg/services/invoice/process_tpps_paid_invoice_report.go]: Added

File: pkg/services/invoice/process_tpps_paid_invoice_report.go (hunk 1 above)
Comment (traskowskycaci) [reply pkg/services/invoice/process_tpps_paid_invoice_report.go:105]: Added

File: pkg/services/invoice/process_edi824.go [hunk 3]
```diff
@@ -163,7 +163,8 @@ func (e *edi824Processor) ProcessFile(appCtx appcontext.AppContext, _ string, st
 			paymentRequest.Status = models.PaymentRequestStatusEDIError
 			err = txnAppCtx.DB().Update(&paymentRequest)
 			if err != nil {
-				txnAppCtx.Logger().Error("failure updating payment request status:", zap.Error(err))
+				txnAppCtx.Logger().Error(fmt.Sprintf("failure updating payment request status : %v", paymentRequest.ID))
+
```
Comment (cameroncaci) [inline pkg/services/invoice/process_edi824.go:167]: Could you maintain the zap error too?

`txnAppCtx.Logger().Error(fmt.Sprintf("failure updating payment request status : %v", paymentRequest.ID), zap.Error(err))`

**Review by cameroncaci — CHANGES_REQUESTED** [review]

File: pkg/services/invoice/process_edi824.go (hunk 3 above)
Comment (traskowskycaci) [reply pkg/services/invoice/process_edi824.go]: Fixed!

**Pull request merged**